src/Tasks/Microsoft.NET.Build.Tasks/GenerateBundle.cs

```diff
@@ -37,14 +37,29 @@ public class GenerateBundle : TaskBase
         protected override void ExecuteCore()
         {
             OSPlatform targetOS = RuntimeIdentifier.StartsWith("win") ? OSPlatform.Windows :
-                                  RuntimeIdentifier.StartsWith("osx") ? OSPlatform.OSX : OSPlatform.Linux;
+                                  RuntimeIdentifier.StartsWith("osx") ? OSPlatform.OSX :
+                                  RuntimeIdentifier.StartsWith("linux") ? OSPlatform.Linux :
+                                  throw new ArgumentException(nameof (RuntimeIdentifier));
+
+            Architecture targetArch = RuntimeIdentifier.EndsWith("x64") ? Architecture.X64 :
+                                      RuntimeIdentifier.EndsWith("x86") ? Architecture.X86 :
+                                      RuntimeIdentifier.EndsWith("arm64") ? Architecture.Arm64 :
+                                      RuntimeIdentifier.EndsWith("arm") ? Architecture.Arm :
+                                      throw new ArgumentException(nameof (RuntimeIdentifier));
```

Comment:
This is problematic - for example `win81-arm64-aot` is a valid RID - which would throw here. Can't we get the target architecture directly from the MSBuild as a parameter/property?

Reply:
Good to know. What are the `aot` rids, and where can I find out more about them? :)

There is a [`PlatformTarget`](https://github.com/dotnet/sdk/blob/master/src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.RuntimeIdentifierInference.targets#L68) computed from the RID. However it [doesn't](https://github.com/dotnet/sdk/blob/master/src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.RuntimeIdentifierInference.targets#L90) have a case for ARM64, even though it is [used] elsewhere (https://github.com/dotnet/sdk/blob/master/src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.RuntimeIdentifierInference.targets#L129). The linker [targets](https://github.com/dotnet/sdk/blob/master/src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.ILLink.targets#L185) might also be incorrect. @wli3 @dsplaisted do you have a recommendation on how to get the target architecture?

Reply:
We should fix PlatformTarget - this comment is obviously wrong: https://github.com/dotnet/sdk/blob/master/src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.RuntimeIdentifierInference.targets#L90

Reply:
That said for 5.0 I think it would be OK to implement a similar algorithm as the SDK has (so EndsWith or Contains) and call it good enough. For 6.0 we should definitely teach the SDK about ARM64 (since it's a perfectly valid platform) and make sure we have ideally only one place which computes the target architecture.

Reply:
There's also [`Platform`](https://github.com/dotnet/sdk/blob/master/src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Sdk.props#L61) which sets `PlatformTarget` but again doesn't have an ARM64 case. And MSBuild has a [`ProcessorArchitecture`](https://github.com/dotnet/msbuild/blob/master/src/Tasks/Microsoft.Common.CurrentVersion.targets#L468) which comes from `PlatformTarget`.

I agree we should fix `PlatformTarget` but am hesitant to touch it in 5.0 - so I'll just fix the RID parsing here unless SDK folks would prefer something else.

I'm curious what the `aot` RIDs are for, but had trouble finding out much about them. Is there a document somewhere?

Reply:
If you want to use the `RuntimeIdentifier` the correct way to do it (eventually) is probably to pass the runtime graph (a file now included in the SDK) to the task, and then use graph operations to figure out if the current RuntimeIdentifier inherits from a more generic RuntimeIdentifier specifying the architecture you're interested in.

Also, per [this](https://github.com/dotnet/project-system/blob/0189463807109508d2de9adf028a8ce81daca69a/docs/repo/coding-conventions.md#data), we should use `PlatformTarget` instead of `Platform` in this type of logic, as `Platform` can be an arbitrary alias that the developer defines.

Reply:
The AOT RIDs are for AOT (Ahead of time compilers) - one such example is building for .NET Native. I don't think they're "widely" used for now.

I don't think bundler should be in the business of trying to understand RIDs - that's a complicated topic. SDK needs to correctly define properties describing the target - in this case `PlatformTarget` - and bundler should just use them. (That is obviously the ideal state, not what we need to do for 5.0)